#VRML_SIM R2022a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.35975279351257 -0.08269903530216759 0.9293754876905206 3.5434026777690586
  position 0.29510548808013864 0.14410394605398083 0.2299652312576639
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0 0 -0.12
  rotation 0 1 0 0
  floorSize 5 5
}
Robot {
  translation -4.18862e-18 1.55614e-18 -0.090045
  rotation -2.698443119727994e-07 -1.1110100315257515e-07 0.9999999999999574 2.9489041081851354e-10
  children [
    DEF left_1_leg Solid {
      translation -0.03 0.04 0
      rotation 0 0 1 -1.5707953071795862
      children [
        DEF left_1_leg_Joint Solid {
          rotation -1.2697596473170847e-07 -0.9992417224549665 -0.038935589185420144 -5.307179586466759e-06
          children [
            DEF left_1_Joint Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
              }
              geometry Cylinder {
                height 0.01
                radius 0.01
              }
            }
          ]
          name "left_1_leg_Joint Solid"
          boundingObject USE left_1_Joint
          physics Physics {
          }
        }
        HingeJoint {
          jointParameters HingeJointParameters {
            position 0.3321633507071567
            axis 0 0 1
          }
          device [
            RotationalMotor {
              name "left_1 motor1"
            }
          ]
          endPoint Solid {
            translation -0.02 8.673617379884035e-19 0
            children [
              DEF left_1_feet_Joint Solid {
                translation -0.02 0 6.74634e-14
                rotation 1 0 0 1.5708
                children [
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      position 0.9692799999999999
                      axis 0 0 1
                    }
                    device [
                      RotationalMotor {
                        name "left_1 motor2"
                      }
                    ]
                    endPoint DEF left_1_feet Solid {
                      translation 0.016489581217938833 -0.01131784923284519 0
                      rotation 0 0 1 0.9692799999999999
                      children [
                        Solid {
                          translation -2.65359e-08 -0.01 0
                          rotation 0.9999999999943767 -8.588329999951705e-07 3.2417399999817706e-06 -1.5707953071795862
                          children [
                            DEF left_1_feet Shape {
                              appearance PBRAppearance {
                                baseColor 0 0 1
                              }
                              geometry Sphere {
                                radius 0.005
                              }
                            }
                          ]
                          name "left_1_feet"
                          boundingObject USE left_1_feet
                          physics Physics {
                          }
                        }
                        DEF left_1_feet Shape {
                          appearance PBRAppearance {
                            baseColor 0 0 1
                          }
                          geometry Box {
                            size 0.005 0.02 0.005
                          }
                        }
                      ]
                      name "left_1_feet"
                      boundingObject USE left_1_feet
                      physics Physics {
                      }
                    }
                  }
                  DEF left_1_feet_Joint Solid {
                    children [
                      DEF left_1_feet_Joint Shape {
                        appearance PBRAppearance {
                          baseColor 1 0 0
                        }
                        geometry Cylinder {
                          height 0.01
                          radius 0.01
                        }
                      }
                    ]
                    name "left_1_feet_Joint Solid"
                    boundingObject USE left_1_feet_Joint
                    physics Physics {
                    }
                  }
                ]
                boundingObject USE left_1_feet_Joint
                physics Physics {
                }
              }
              DEF left_1_leg Shape {
                appearance PBRAppearance {
                  baseColor 0 0 1
                }
                geometry Box {
                  size 0.04 0.005 0.005
                }
              }
            ]
            name "left_1_leg"
            boundingObject USE left_1_leg
            physics Physics {
            }
          }
        }
      ]
      name "left_1_leg Solid(1)"
      boundingObject USE left_1_leg
      physics Physics {
        density 500
      }
    }
    DEF left_2_leg Solid {
      translation 0.03 0.04 0
      rotation 0 0 1 -1.5707953071795862
      children [
        DEF left_2_leg_Joint Solid {
          rotation 0 1 0 5.307156870259904e-06
          children [
            DEF left_1_Joint Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
              }
              geometry Cylinder {
                height 0.01
                radius 0.01
              }
            }
          ]
          name "left_1_leg_Joint Solid"
          boundingObject USE left_1_Joint
          physics Physics {
          }
        }
        HingeJoint {
          jointParameters HingeJointParameters {
            position 0.3321633507071567
            axis 0 0 1
          }
          device [
            RotationalMotor {
              name "left_2 motor1"
            }
          ]
          endPoint Solid {
            translation -0.02 8.673617379884035e-19 0
            children [
              DEF left_2_feet_Joint Solid {
                translation -0.02 0 6.74634e-14
                rotation 1 0 0 1.5708
                children [
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      position 0.9692799999999999
                      axis 0 0 1
                    }
                    device [
                      RotationalMotor {
                        name "left_2 motor2"
                      }
                    ]
                    endPoint DEF left_2_feet Solid {
                      translation 0.016489581217938833 -0.01131784923284519 0
                      rotation 0 0 1 0.9692799999999999
                      children [
                        Solid {
                          translation -2.65359e-08 -0.01 0
                          rotation 0.9999999999943767 -8.588329999951705e-07 3.2417399999817706e-06 -1.5707953071795862
                          children [
                            DEF left_2_feet Shape {
                              appearance PBRAppearance {
                                baseColor 0 0 1
                              }
                              geometry Sphere {
                                radius 0.005
                              }
                            }
                          ]
                          name "left_2_feet"
                          boundingObject USE left_2_feet
                          physics Physics {
                          }
                        }
                        DEF left_2_feet Shape {
                          appearance PBRAppearance {
                            baseColor 0 0 1
                          }
                          geometry Box {
                            size 0.005 0.02 0.005
                          }
                        }
                      ]
                      name "left_2_feet"
                      boundingObject USE left_2_feet
                      physics Physics {
                      }
                    }
                  }
                  DEF left_2_feet_Joint Solid {
                    children [
                      DEF left_1_feet_Joint Shape {
                        appearance PBRAppearance {
                          baseColor 1 0 0
                        }
                        geometry Cylinder {
                          height 0.01
                          radius 0.01
                        }
                      }
                    ]
                    name "left_1_feet_Joint Solid"
                    boundingObject USE left_1_feet_Joint
                    physics Physics {
                    }
                  }
                ]
                boundingObject USE left_1_feet_Joint
                physics Physics {
                }
              }
              DEF left_2_leg Shape {
                appearance PBRAppearance {
                  baseColor 0 0 1
                }
                geometry Box {
                  size 0.04 0.005 0.005
                }
              }
            ]
            name "left_2_leg"
            boundingObject USE left_2_leg
            physics Physics {
            }
          }
        }
      ]
      name "left_2_leg Solid"
      boundingObject USE left_2_leg
      physics Physics {
        density 500
      }
    }
    DEF Right_1_leg Solid {
      translation -0.03 -0.04 0
      rotation 0 0 1 1.5708
      children [
        DEF Right_1_leg_Joint Solid {
          rotation 0 1 0 5.307156870259904e-06
          children [
            DEF left_1_Joint Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
              }
              geometry Cylinder {
                height 0.01
                radius 0.01
              }
            }
          ]
          name "left_1_leg_Joint Solid"
          boundingObject USE left_1_Joint
          physics Physics {
          }
        }
        HingeJoint {
          jointParameters HingeJointParameters {
            position 0.3321633507071567
            axis 0 0 1
          }
          device [
            RotationalMotor {
              name "Right_1 motor1"
            }
          ]
          endPoint Solid {
            translation -0.02 8.673617379884035e-19 0
            children [
              DEF Right_1_feet_Joint Solid {
                translation -0.02 0 6.74634e-14
                rotation 1 0 0 1.5708
                children [
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      position 0.9692799999999999
                      axis 0 0 1
                    }
                    device [
                      RotationalMotor {
                        name "Right_1 motor2"
                      }
                    ]
                    endPoint DEF Right_1_feet Solid {
                      translation 0.016489581217938833 -0.01131784923284519 0
                      rotation 0 0 1 0.9692799999999999
                      children [
                        Solid {
                          translation -2.65359e-08 -0.01 0
                          rotation 0.9999999999943767 -8.588329999951705e-07 3.2417399999817706e-06 -1.5707953071795862
                          children [
                            DEF Right_1_feet Shape {
                              appearance PBRAppearance {
                                baseColor 0 0 1
                              }
                              geometry Sphere {
                                radius 0.005
                              }
                            }
                          ]
                          name "Right_1_feet"
                          boundingObject USE Right_1_feet
                          physics Physics {
                          }
                        }
                        DEF Right_1_feet Shape {
                          appearance PBRAppearance {
                            baseColor 0 0 1
                          }
                          geometry Box {
                            size 0.005 0.02 0.005
                          }
                        }
                      ]
                      name "Right_1_feet"
                      boundingObject USE Right_1_feet
                      physics Physics {
                      }
                    }
                  }
                  DEF Right_1_feet_Joint Solid {
                    children [
                      DEF left_1_feet_Joint Shape {
                        appearance PBRAppearance {
                          baseColor 1 0 0
                        }
                        geometry Cylinder {
                          height 0.01
                          radius 0.01
                        }
                      }
                    ]
                    name "left_1_feet_Joint Solid"
                    boundingObject USE left_1_feet_Joint
                    physics Physics {
                    }
                  }
                ]
                boundingObject USE left_1_feet_Joint
                physics Physics {
                }
              }
              DEF Right_1_leg Shape {
                appearance PBRAppearance {
                  baseColor 0 0 1
                }
                geometry Box {
                  size 0.04 0.005 0.005
                }
              }
            ]
            name "Right_1_leg"
            boundingObject USE Right_1_leg
            physics Physics {
            }
          }
        }
      ]
      name "Right_1_leg Solid"
      boundingObject USE Right_1_leg
      physics Physics {
        density 500
      }
    }
    DEF Right_2_leg Solid {
      translation 0.0300002 -0.04 0
      rotation 0 0 1 1.5708
      children [
        DEF Right_2_leg_Joint Solid {
          rotation 0 1 0 5.307156870259904e-06
          children [
            DEF Right_2_Joint Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
              }
              geometry Cylinder {
                height 0.01
                radius 0.01
              }
            }
          ]
          name "Right_2_leg_Joint Solid"
          boundingObject USE Right_2_Joint
          physics Physics {
          }
        }
        HingeJoint {
          jointParameters HingeJointParameters {
            position 0.3321633507071567
            axis 0 0 1
          }
          device [
            RotationalMotor {
              name "Right_2 motor1"
            }
          ]
          endPoint Solid {
            translation -0.02 8.673617379884035e-19 0
            children [
              DEF left_2_feet_Joint Solid {
                translation -0.02 0 6.74634e-14
                rotation 1 0 0 1.5708
                children [
                  HingeJoint {
                    jointParameters HingeJointParameters {
                      position 0.9692799999999999
                      axis 0 0 1
                    }
                    device [
                      RotationalMotor {
                        name "Right_2 motor2"
                      }
                    ]
                    endPoint DEF left_1_feet Solid {
                      translation 0.016489581217938833 -0.01131784923284519 0
                      rotation 0 0 1 0.9692799999999999
                      children [
                        Solid {
                          translation -2.65359e-08 -0.01 0
                          rotation 0.9999999999943767 -8.588329999951705e-07 3.2417399999817706e-06 -1.5707953071795862
                          children [
                            DEF Right_2_feet Shape {
                              appearance PBRAppearance {
                                baseColor 0 0 1
                              }
                              geometry Sphere {
                                radius 0.005
                              }
                            }
                          ]
                          name "Right_2_feet"
                          boundingObject USE Right_2_feet
                          physics Physics {
                          }
                        }
                        DEF Right_2_feet Shape {
                          appearance PBRAppearance {
                            baseColor 0 0 1
                          }
                          geometry Box {
                            size 0.005 0.02 0.005
                          }
                        }
                      ]
                      name "Right_2_feet"
                      boundingObject USE Right_2_feet
                      physics Physics {
                      }
                    }
                  }
                  DEF left_2_feet_Joint Solid {
                    children [
                      DEF left_2_feet_Joint Shape {
                        appearance PBRAppearance {
                          baseColor 1 0 0
                        }
                        geometry Cylinder {
                          height 0.01
                          radius 0.01
                        }
                      }
                    ]
                    name "left_1_feet_Joint Solid"
                    boundingObject USE left_2_feet_Joint
                    physics Physics {
                    }
                  }
                ]
                boundingObject USE left_2_feet_Joint
                physics Physics {
                }
              }
              DEF Right_2_leg Shape {
                appearance PBRAppearance {
                  baseColor 0 0 1
                }
                geometry Box {
                  size 0.04 0.005 0.005
                }
              }
            ]
            name "Right_2_leg"
            boundingObject USE Right_2_leg
            physics Physics {
            }
          }
        }
      ]
      name "Right_2_leg Solid"
      boundingObject USE Right_2_leg
      physics Physics {
        density 500
      }
    }
    DEF BODY Shape {
      appearance Appearance {
        material Material {
          diffuseColor 1 1 0
        }
        name "body"
      }
      geometry Box {
        size 0.09 0.07 0.02
      }
    }
  ]
  boundingObject USE BODY
  physics Physics {
  }
  controller "my_controller"
}
